hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -567,6 +564,16 @@ private synchronized void close(Pair<String, String> partitionFileSlicePair) {
     closeReader(readers);
   }
 
+  /**
+   * Close and clear all the partitions readers.
+   */
+  private void closePartitionReaders() {
```

Comment:
No, it is related, because in this patch, we refresh the metadata table when the fs view is refreshed, and there are some states in the readers we need to clear about.